h2/src/main/org/h2/value/ValueStringFixed.java

```diff
@@ -42,7 +70,14 @@ public int getType() {
      * @return the value
      */
     public static ValueStringFixed get(String s) {
-        s = trimRight(s);
+        return get(s, PRECISION_TRIM, null);
+    }
+    public static ValueStringFixed get(String s, int precision, Mode mode) {
+        if (mode != null && mode.padFixedStrings && precision < Integer.MAX_VALUE) {
```

Comment:
It will be false in the case of CHAR types without a length specification. In that case the precision of the Column (from which precision comes) will be Integer.MAX_VALUE. This is to mitigate that there will not be any padding in that case.